src/main/scala/akka/persistence/cassandra/journal/PubSubThrottler.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.persistence.cassandra.journal
+
+import scala.concurrent.duration.FiniteDuration
+
+import akka.actor.{ Actor, ActorRef }
+
+/**
+ * Proxies messages to another actor, only allowing identical messages through once every [interval].
+ */
+class PubSubThrottler(delegate: ActorRef, interval: FiniteDuration) extends Actor {
+  import PubSubThrottler._
+  import context.dispatcher
+
+  val intervalMs = interval.toMillis
+  var seen = Map.empty[Any, Long]
+  context.system.scheduler.schedule(interval, interval, self, Cleanup)
+
+  def receive = {
+    case Cleanup =>
+      seen = seen.filterNot { case (msg, time) => expired(time) }
+
+    case msg =>
+      if (seen.get(msg).filterNot(expired).isEmpty) {
```

Comment:
`filterNot` might be expensive, do we really need to track the timestamp of each message? wouldn't it be more efficient (and good enough) to use a `Set` for the incoming messages. Send immediately if not already in the `Set`. Flush (send) the messages in the `Set` in the periodic task. Might need a second `Set` (or a wrapper) to avoid sending same twice if it is not received again during the interval.